satpy/readers/fci_l2_nc.py

```diff
@@ -0,0 +1,247 @@
+            logger.warning("Start time cannot be obtained from file content, using default value instead")
+            start_time = datetime.strptime('20200101120000', '%Y%m%d%H%M%S')
+        return start_time
+
+    @property
+    def end_time(self):
+        """Get observation end time."""
+        try:
+            end_time = datetime.strptime(self.nc.attrs['time_coverage_end'], '%Y%m%d%H%M%S')
+        except (ValueError, KeyError):
+            # TODO if the sensing_end_time_utc attribute is not valid, adds 20 minutes to the start time
+            end_time = self.start_time + timedelta(minutes=PRODUCT_DATA_DURATION_MINUTES)
+        return end_time
+
+    @property
+    def spacecraft_name(self):
+        """Return spacecraft name."""
+        try:
+            return self.nc.attrs['platform']
+        except KeyError:
+            # TODO if the platform attribute is not valid, return a default value
+            logger.warning("Spacecraft name cannot be obtained from file content, using default value instead")
+            return 'DEFAULT_MTG'
+
+    @property
+    def sensor(self):
+        """Return instrument."""
+        try:
+            return self.nc.attrs['data_source']
+        except KeyError:
+            # TODO if the data_source attribute is not valid, return a default value
+            logger.warning("Sensor cannot be obtained from file content, using default value instead")
+            return 'fci'
+
+    @property
+    def ssp_lon(self):
+        """Return subsatellite point longitude."""
+        try:
+            return float(self._projection.attrs['longitude_of_projection_origin'])
+        except KeyError:
+            # TODO if the longitude_of_projection_origin attribute is not valid, return a default value
+            logger.warning("ssp_lon cannot be obtained from file content, using default value instead")
+            return 0.
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get dataset using the file_key in dataset_info."""
+        var_key = dataset_info['file_key']
+        logger.debug('Reading in file to get dataset with key %s.', var_key)
+
+        try:
+            fci_variable = self.nc[var_key]
+        except KeyError:
+            logger.warning("Could not find key %s in NetCDF file, no valid Dataset created", var_key)
+            return None
+
+        # Put the values in a dask array
+        data_values_dask = da.from_array(fci_variable, CHUNK_SIZE)
+
+        # TODO in some of the test files, invalid pixels contain the value defined as "fill_value" in the YAML file
+        # TODO instead of being masked directly in the netCDF variable.
+        # TODO therefore NaN is applied where such value is found or (0 if the array contains integer values)
+        # TODO the next 4 lines have to be removed once the product files are correctly configured
+        try:
+            data_values_dask[data_values_dask == dataset_info['fill_value']] = np.NaN
```

Comment:
it's best to work with xarray.DataArray as you can use the `where` method instead.